Sound Open Firmware 2.1.1 Userspace tools + firmware blobs

cthbleachbit opened the pull request:
Topic Description
-----------------

This PR introduces [Sound Open Firmware](https://www.sofproject.org/) packages. SOF is required to drive audio I/O on 10th gen and newer Intel processors.

Alsa components work as usual but sof components are untested due to lack of compatible hardware.

Package(s) Affected
-------------------

- `alsa-lib` v1.2.6.1
- `alsa-utils` v1.2.6+git20220510
- `audio-base` v0
- `sof-tools` v2.1.1
- `sof-firmware` v2.1.1

Build Order
-----------

```
sof-tools sof-firmware alsa-lib alsa-util audio-base
```

Test Build(s) Done
------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

cthbleachbit committed:
alsa-utils: upgrade to 1.2.6+git20220510

alsatplg now supports `-I' option to add include paths, used by sof
userspace programs to generate topology descriptors.

cthbleachbit committed:
alsa-lib: upgrade to 1.2.6.1

cthbleachbit committed:
sof-userspace: new, 2.1.1

cthbleachbit committed:
sof-firmware: new, 2.1.1

MingcongBai commented on extra-admin/sof-userspace/02-tplg/build:
```diff
@@ -0,0 +1,7 @@
+DESTDIR="${PKGDIR}/usr/lib/firmware/intel/sof-tplg-v${PKGVER}"
+
+mkdir -p "${DESTDIR}"
```
Missing verbosity.

MingcongBai commented on extra-admin/sof-userspace/02-tplg/build:
```diff
@@ -0,0 +1,7 @@
+mkdir -p "${DESTDIR}"
+
+for i in "${SRCDIR}"/abbuild/topology/*.tplg; do
+	install -vm644 "$i" -t "${DESTDIR}"
```
Missing abinfo.

MingcongBai commented on extra-kernel/sof-firmware/autobuild/build:
```diff
@@ -0,0 +1,6 @@
+FW_DEST="${PKGDIR}/usr/lib/firmware/intel"
```
Missing abinfo throughout.

MingcongBai commented on extra-kernel/sof-firmware/autobuild/build:
```diff
@@ -0,0 +1,6 @@
+
+rsync -va "${SRCDIR}"/sof-v"$PKGVER" "${FW_DEST}"/
+
+install -Dvt "${PKGDIR}/usr/share/doc/${PKGNAME}" -m644 LICENCE* Notice*
```
Man this parameter arrangement makes my head spin.

MingcongBai commented on extra-multimedia/alsa-utils/autobuild/prepare:
```diff
@@ -0,0 +1 @@
+autoreconf -fi
```
Missing abinfo, verbosity. Also, why would this be necessary, was RECONF disabled?

MingcongBai commented on extra-multimedia/alsa-utils/spec:
```diff
@@ -1,4 +1,5 @@
-SRCS="tbl::https://www.alsa-project.org/files/pub/utils/alsa-utils-$VER.tar.bz2"
-CHKSUMS="sha256::9c169ae37a49295f9b97b92ace772803daf6b6510a19574e0b78f87e562118d0"
+_BASE_VER=1.2.6
+VER=${_BASE_VER}+git20220510
```
Why?

MingcongBai replied:
Scratch that, but I'd suggest backporting specific patches instead of switching source format - perhaps see if you could backport a series of patches from `topology/` since December 6, 2021.

MingcongBai committed:
sof-userspace: only build sof-tools

- soft-tplg is provided as part of the sof-bin repository, and upstream
  recommends installing both firmware and topology (tplg) data with
  their install.sh script.
- Rename package path to extra-admin/sof-tools

MingcongBai committed:
sof-firmware: improve build script

- Use upstream-provided install.sh to install firmware and topology data.
- Lint scripts in accordance with the Styling Manual.

MingcongBai committed:
alsa-utils: add abinfo, verbosity to prepare

MingcongBai committed:
audio-base: new, 0

MingcongBai committed:
sof-firmware: fix a typo in build script

MingcongBai committed:
sof-tools: move to extra-multimedia

MingcongBai commented:
Tested on Panasonic Let's Note CF-SV2M (11th Gen Intel platform). Installing `audio-base` and rebooting resulted in working audio.
